[Agentic Search]Documentation with all the features (#11138)
opensearch-project/documentation-website: _vector-search/ai-search/agentic-search/flow-agent.md

---
layout: default
title: Flow agent
parent: Agentic search
grand_parent: AI search
nav_order: 110
has_children: false
---

# Flow agent

> The Conversational agent provides great flexibility and supports many use cases. However, you might not need all that functionality for every scenario. You can directly invoke the Query Planner Tool using a flow agent, which reduces the number of LLM calls made, thereby reducing both cost and latency.

Reviewer: ```suggestion
The Conversational agent provides great flexibility and supports many use cases. However, if you do need all that functionality for every scenario. You can directly invoke the Query Planner Tool using a flow agent, which reduces the number of LLM calls made, thereby reducing both cost and latency.
```

                    "description": "Classic canvas sneakers",
                    "price": 65.0,
                    "currency": "USD",
                    "rating": 4.2,
                    "review_count": 2100,
                    "in_stock": true,
                    "color": "white",
                    "size": "8",
                    "category": "shoes",
                    "brand": "Converse",
                    "tags": [
                        "casual",
                        "classic",
                        "canvas"
                    ]
                },
                "sort": [
                    4.2,
                    2100
                ]
            }
        ]
    },
    "ext": {
        "dsl_query": "{\"size\":10.0,\"query\":{\"bool\":{\"filter\":[{\"term\":{\"category\":\"shoes\"}},{\"term\":{\"color\":\"white\"}},{\"range\":{\"price\":{\"lt\":150.0}}}]}},\"sort\":[{\"rating\":{\"order\":\"desc\"}},{\"review_count\":{\"order\":\"desc\"}}]}"
    }
}
```

**Use Flow Agent when:**
- Latency is critical
- Cost optimization is important
- Simple queries are sufficient
- No conversation memory needed

> **Use Conversational Agent when:**

Reviewer: - Intelligent Search
 Add this too
